src/training/tesstrain_utils.sh

```diff
@@ -83,9 +83,7 @@ run_command() {
     fi
     shift
     tlog "[$(date)] ${cmd} $@"
-    "${cmd}" "$@" 2>&1 1>&2 | tee -a ${LOG_FILE}
-    # check completion status
-    if [[ $? -gt 0 ]]; then
+    if ! "${cmd}" "$@" 2>&1 1>&2 | tee -a ${LOG_FILE}; then
```

Comment:
```suggestion
    if ! "${cmd}" "$@" 2>&1 | tee -a ${LOG_FILE}; then
```

Reply:
yes, I had overlooked that. And even worse, prior to that, `which` status was also not working (see 
91a7e86).